conda_lock/common.py

```diff
@@ -43,6 +55,21 @@ def write_file(obj: str, filepath: Union[str, pathlib.Path]) -> None:
         fp.write(obj)
 
 
+@contextmanager
+def temporary_file_with_contents(content: str) -> Iterator[pathlib.Path]:
+    """Generate a temporary file with the given content.  This file can be used by subprocesses
+
+    On Windows, NamedTemporaryFiles can't be opened a second time, so we have to close it first (and delete it manually later)
```

Comment:
you can create a ``tempfile.TemporaryDirectory`` and put the file there, after it leaves the context manager the folder will be deleted